client_consumer/pulp/client/consumer/cli.py

```diff
@@ -91,43 +91,39 @@ def initialize(context):
     context.cli.add_command(StatusCommand(context, 'status', _(d)))
 
 
-def download(url, location):
+def write_to_location(location, content):
     """
-    Download files to the specified location.
-    :param url: The file URL.
-    :type url: str
-    :param location: The absolute path to where the downloaded
-        file is to be stored.
-    :type location: str
+    Write content to a path. Ensures that the entire path exists, creating directories if necessary.
+
+    :param location: path that should exist
+    :type  location: str
+    :param content: bits to be written to file
+    :type  content: str
     """
-    request = urllib2.urlopen(url)
     try:
-        content = request.read()
-        fp = open(location, 'w+')
-        try:
-            fp.write(content)
-        finally:
-            fp.close()
+        os.makedirs(os.path.dirname(location))
+    except OSError, e:
+        if e.errno != errno.EEXIST:
+            raise
+    fp = open(location, 'w+')
+    try:
+        fp.write(content)
     finally:
-        request.close()
+        fp.close()
 
 
-def update_server_key(conf):
+def update_server_key(self):
```

Comment:
This appears to be a module-level function. What is self? I recommend not using the name self here because of the convention for using it to reference a class instance on methods.